lib/account/account-service.js

```diff
@@ -1,70 +1,74 @@
 "use strict";
 
 /**
- * What we initially send back to the client so they can start to initiate
- * the complete procedure and save information.
- *
- * @typedef {Object} AccountService~initiateResult
- * @property {string} accountId
- * @property {string} email
- * @property {string} mfaKey
- * @property {string} salt
+ * Handles the creating, deletion, and updating of account and
+ * registration files.
  */
 
 /**
- * Stored information for the account.  This is changed from the initiated
- * file as we add password and there could be other items added after
- * we initiate the account.
+ * Information about an account holder. Used to generate/validate
+ * a password and two factor authentication as well as Ids and
+ * contact information.
+ *
+ * Once the account is complete, regId is removed.
  *
  * @typedef {Object} AccountService~accountFile
- * @property {string} accountId
  * @property {string} email
  * @property {string} mfaKey
  * @property {string} password
- * @property {string} salt
+ * @property {string} passwordSalt
  */
 
 /**
- * Handles account creation from initiating to confirmation.
+ * @param {Object} config
+ * @param {Object} secureHash
+ * @param {Object} Storage
+ * @return {Object} account service object
  */
-class AccountService {
-    /**
-     * Used for Dependency Injection and initial configuration
-     * of the storage plugin.
-     *
-     * @param {Object} config
-     * @param {Object} password
-     * @param {Object} Storage
-     */
-    constructor(config, secureHash, Storage) {
-        this.accountConfig = config.account || {};
-        this.directoryName = this.accountConfig.accountDir || "account/";
-        this.secureHash = secureHash;
-        this.storage = Storage.configure(config.storage);
-    }
+module.exports = function (config, secureHash, Storage) {
+    var accountConfig, accountDirName, registrationDirName, storage;
 
+    accountConfig = config.account || {};
+    accountDirName = accountConfig.accountDir || "account/";
+    registrationDirName = accountConfig.registrationDir || "registration/";
+    storage = new Storage().configure(config.storage);
```

Comment:
Shouldn't we have "storage" injected instead of "Storage"?

Please consider this and maybe we can come up with something.